Helper/Url.php

```diff
@@ -233,11 +306,82 @@ public function getPreviewUrlFront(Store $store)
         $url = $this->urlBuilder->getUrl(
             '',
             [
-                '_nosid' => true,
-                '_scope_to_url' => true,
-                '_scope' => $store->getCode(),
+                self::MAGENTO_URL_OPTION_NOSID => true,
+                self::MAGENTO_URL_OPTION_SCOPE_TO_URL => true,
+                self::MAGENTO_URL_OPTION_SCOPE => $store->getCode(),
             ]
         );
         return $this->addNostoDebugParamToUrl($url);
     }
+
+    /**
+     * Gets the absolute URL to the current store view cart page.
+     *
+     * @param Store $store the store to get the url for.
+     * @param string $currentUrl restore cart url
+     * @return string the url.
+     */
+    public function getUrlCart(Store $store, $currentUrl)
+    {
+        // @codingStandardsIgnoreLine
+        $userQuery = parse_url($currentUrl, PHP_URL_QUERY);
+        if (is_string($userQuery)) {
+            parse_str($userQuery, $urlParameters); // @codingStandardsIgnoreLine
+        } else {
+            $urlParameters = [];
+        }
+
+        $defaultParams = $this->getUrlOptionsWithNoSid();
+        $url = $store->getUrl(
+            self::MAGENTO_PATH_CART,
+            $defaultParams
+        );
+
+        if (!empty($urlParameters)) {
+            foreach ($urlParameters as $key => $val) {
+                $url = HttpRequest::replaceQueryParamInUrl(
+                    $key,
+                    $val,
+                    $url
+                );
+            }
+        }
+
+        return $url;
+    }
+
+    /**
+     * Returns restore cart url
+     *
+     * @param string $hash
+     * @param Store $store
+     * @return string
+     */
+    public function generateRestoreCartUrl($hash, Store $store)
```

Comment:
I would move this method into it's own class `Nosto\Tagging\Model\Cart\Restore\Builder.php`